src/LtiDeepLinkResourceIcon.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Packback\Lti1p3;
+
+class LtiDeepLinkResourceIcon
+{
+    private $url;
+    private $width;
+    private $height;
+
+    public function __construct(string $url, ?int $width = null, ?int $height = null)
```

Comment:
Is `height` and `width` truly optional? In reading through the IMS docs, it doesn't indicate whether they are, but every example includes them. Have you tested not setting these values but only providing a URL? If so, what happened?

Reply:
if it _is_ required, I would also note that `public function setHeight(?int $height)` and `public function setWidth(?int $width)` should be changed to make those arguments required.

Reply:
Well I'm not really sure because the lti 1.3 docs are lacking a lot of information and it probably depends on how the platform handles the icons. The url was the only thing I was sure about because without a url the object doesn't make sense :smile: But as all examples in the IMS docs contain width and height I would agree that it's probably better to make them required.